x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/PolicyStepsRegistry.java

```diff
@@ -214,7 +232,8 @@ private List<Step> getAllStepsForIndex(ClusterState state, Index index) {
             ClientHelper.INDEX_LIFECYCLE_ORIGIN,
             policyMetadata.getHeaders()
         );
-        return policyMetadata.getPolicy().toSteps(policyClient, licenseState);
+        return policyMetadata.getPolicy()
+            .toSteps(policyClient, indexMetadata.getLifecycleExecutionState().actionsOrderVersion(), licenseState);
```

Comment:
I know it's extremely unlikely, but could we possibly reach this and have `indexMetadata.getLifecycleExecutionState()` return null and hit an NPE?